### ISSUE TITLE
Request to perform gain phase averaging on secondary calibrators

### ISSUE THREAD

**Issue by Sarah-Bradbury** [AlecThomson/quocka#2]
This is a request to perform antenna gain phase averaging (using the `gpaver` task in miriad) on the secondary calibrators during the primary calibration stages. The change would occur in the 'run_cal.py' script. Currently, the antenna gain solutions are not being averaged prior to the `gpcopy` step. 

The code currently reads (from the 'run_cal.py' script):
```python
for t in targetnames:
            logger.info(
                "Working on source %s" % t,
            )
            # Move on to the target!
            call(
                ["gpcopy", "vis=%s" % seccalname, "out=%s" % t],
            )
            flag(
                t,
            )
            flag(
                t,
            )
```

To perform the gain phase averaging on secondary calibrators, the code above could be changed to:
```python
for t in targetnames:
            logger.info(
                "Working on source %s" % t,
            )
            # Move on to the target!
            logger.info(
                "Averaging secondary cal gain solutions over 10 min interval..."
            )
            call(
                ["gpaver", "interval=10", "vis=%s" % seccalname, "options=scalar"],    
            )
            call(
                ["gpcopy", "vis=%s" % seccalname, "out=%s" % t],
            )
            flag(
                t,
            )
            flag(
                t,
            )
 ```
Thanks again!

**Comment by AlecThomson** [comment]
Thanks @Sarah-Bradbury ! I think this should be fairly straight forward to implement. I wonder - do we want the ability to set the averaging interval from the config file? Otherwise it might be good to document why 10mins is chosen

**Comment by AlecThomson** [comment]
Hi @Sarah-Bradbury - I've put together a potential solution in #4 . Would you be happy to test it on your end at let me know how it goes?

**Comment by Sarah-Bradbury** [comment]
Hi @AlecThomson, thanks so much for putting a potential solution together. I will give it a go on my end and let you know how it goes.

**Comment by Sarah-Bradbury** [comment]
> 

Hi @AlecThomson, Re: choosing 10 seconds as the averaging interval:
- I think it would be good to have the flexibility to set this value from the config file
- However, I think 10 is a good default value because it's longer than any plausible secondary cal visit, but shorter than any plausible interval between secondary cal visit for these freq bands. In other words, we want to average gain solutions down to a single value per sec cal visit, before the calibration is interpolated and applied to the targets.

**Comment by AlecThomson** [comment]
@Sarah-Bradbury - just be aware there was a bug in my previous implementation (both #4 and #5 ). I accidentally had `gpcopy` before `gpaver`! This is now fixed

**Comment by AlecThomson** [comment]
Implemented in #5
